File: leetcode/python/66.plus-one.py
from typing import List


class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        res = digits[::-1]
        carry = 1
        for i in range(len(res)):
            res[i] += carry
            carry = res[i] // 10
            res[i] = res[i] % 10

        if carry:
            res.append(1)
        return res[::-1]

File: leetcode/cpp/66.plus-one.cpp
#include "oj_header.h"

class Solution {
public:
    vector<int> plusOne(vector<int>& digits)
    {
        int carry = 0;
        deque<int> res;
        digits[digits.size() - 1]++;
        for (int i = digits.size() - 1; i >= 0; i--) {
            int t = digits[i];
            t = carry ? t + 1 : t;
            carry = t / 10;
            t = t % 10;
            res.push_front(t);
        }
        if (carry)
            res.push_front(1);
        return vector<int>(res.begin(), res.end());
    }
};